File: src/Prime.java
public class Prime {
    public static boolean is_prime(int number) {
        if (number < 2) {
            return false;
        }

        if (number == 2 || number == 3) {
            return true;
        }

        if (number % 2 == 0 || number % 3 == 0) {
            return false;
        }

        int i = 5;
        while (i * i <= number) {
            if (number % i == 0 || number % (i + 2) == 0) {
                return false;
            }
            i += 6;
        }
        return true;
    }
}

File: src/prime.rs
macro_rules! is_prime {
    ($number:expr) => {{
        let number = $number;
        if number < 2 {
            false
        } else if number == 2 || number == 3 {
            true
        } else if number % 2 == 0 || number % 3 == 0 {
            false
        } else {
            let mut i: u64 = 5;
            let mut result = true;
            while i * i <= number {
                if number % i == 0 || number % (i + 2) == 0 {
                    result = false;
                    break;
                }
                i += 6;
            }
            result
        }
    }};
}
